.github/workflows/on-merge-branch-changeset-release-main.yml

```diff
@@ -8,6 +8,13 @@ concurrency:
   group: ${{ github.head_ref || github.ref }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
```

Comment:
Workflow permissions are pretty broad by default. So I've restricted each workflow and job to the minimum it needs to do its thing.